stmhal/flash.c

```diff
@@ -83,12 +99,27 @@ static const uint32_t flash_info_table[26] = {
     ADDR_FLASH_SECTOR_10, FLASH_SECTOR_10,
     ADDR_FLASH_SECTOR_11, FLASH_SECTOR_11,
     #endif
+	#if defined(FLASH_SECTOR_12)
+	ADDR_FLASH_SECTOR_12, FLASH_SECTOR_12,
+	ADDR_FLASH_SECTOR_13, FLASH_SECTOR_13,
+	ADDR_FLASH_SECTOR_14, FLASH_SECTOR_14,
+	ADDR_FLASH_SECTOR_15, FLASH_SECTOR_15,
+	ADDR_FLASH_SECTOR_16, FLASH_SECTOR_16,
+	ADDR_FLASH_SECTOR_17, FLASH_SECTOR_17,
+	ADDR_FLASH_SECTOR_18, FLASH_SECTOR_18,
+	ADDR_FLASH_SECTOR_19, FLASH_SECTOR_19,
+	ADDR_FLASH_SECTOR_20, FLASH_SECTOR_20,
+	ADDR_FLASH_SECTOR_21, FLASH_SECTOR_21,
+	ADDR_FLASH_SECTOR_22, FLASH_SECTOR_22,
+	ADDR_FLASH_SECTOR_23, FLASH_SECTOR_23,
+	#endif
     ADDR_FLASH_END, 0,
 };
+static const uint32_t flash_info_table_size = sizeof(flash_info_table)/sizeof(flash_info_table[0]);
```

Comment:
This macro is defined in ../py/misc.h and pulls in mpconfigport.h.
Define the variable with static const results in a file scope, we have type safety and the complier probably replaces the appearance of the constant during compile time by its value. Why would you choose a #define?